docs/about/whats_new.rst

```diff
@@ -16,6 +16,7 @@ v1.8.next
 - Tweak ``list_products`` logic for getting crs and resolution values (:pull:`1535`)
 - Add new ODC Cheatsheet reference doc to Data Access & Analysis documentation page (:pull:`1543`)
 - Fix broken codecov github action. (:pull:`1554`)
+- Fix handling of date value as int in Query construction (:pull:`1561`)
```

Comment:
I think this needs a longer explanation, since it very well could break code that people are running.
State that that ints and floats used to be accepted and converted as dates, but it was never documented and explaining, is unclear how it should work correctly, and so is being removed.